flink-tests-java17/src/test/java/org/apache/flink/api/java/typeutils/runtime/PojoRecordSerializerUpgradeTestSpecifications.java

```diff
@@ -81,18 +80,19 @@ public TypeSerializer<PojoAfterUpgrade> createUpgradedSerializer() {
             TypeSerializer<PojoAfterUpgrade> serializer =
                     TypeExtractor.createTypeInfo(PojoAfterUpgrade.class)
                             .createSerializer(new SerializerConfigImpl());
-            assertSame(PojoSerializer.class, serializer.getClass());
+            assertThat(serializer.getClass()).isSameAs(PojoSerializer.class);
             return serializer;
         }
 
         @Override
-        public Matcher<PojoAfterUpgrade> testDataMatcher() {
-            return is(new PojoAfterUpgrade(911108, "Gordon"));
+        public Condition<PojoAfterUpgrade> testDataCondition() {
+            return new Condition<>(
+                    new PojoAfterUpgrade(911108, "Gordon")::equals, "value is (911108, Gordon");
```

Comment:
```suggestion
                    new PojoAfterUpgrade(911108, "Gordon")::equals, "value is (911108, Gordon)");
```

Reply:
thanks @Zakelly for the quick review!

Updated.